<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="cloud_topic_in_1" default="/d435/depth/color/points"/>
  <arg name="cloud_topic_in_2" default="/d455/depth/color/points"/>
  <arg name="cloud_merged_topic" default="/camera/depth/color/points" />
  <arg name="mode" default="$(env MOVEL_MODE)" doc="mode: [deployment, development]"/>

  <group if="$(eval mode == 'deployment')">
    <node pkg="velocity_limiter" type="point_cloud_merger_node" name="point_cloud_merger" output="screen" respawn="true">
      <rosparam file="$(find velocity_limiter)/config/point_cloud_merger.yaml"/>
      <rosparam file="$(env HOME)/.config/movel/config/velocity_limiter/config/point_cloud_merger.yaml"/>
      <remap from="/cloud_1" to="$(arg cloud_topic_in_1)" />
      <remap from="/cloud_2" to="$(arg cloud_topic_in_2)" />
      <remap from="/cloud/merged" to="$(arg cloud_merged_topic)" />
    </node>
  </group>

  <group unless="$(eval mode == 'deployment')">    
    <node pkg="velocity_limiter" type="point_cloud_merger_node" name="point_cloud_merger" output="screen" respawn="true">
      <rosparam file="$(find velocity_limiter)/config/point_cloud_merger.yaml"/>
      <remap from="/cloud_1" to="$(arg cloud_topic_in_1)" />
      <remap from="/cloud_2" to="$(arg cloud_topic_in_2)" />
      <remap from="/cloud/merged" to="$(arg cloud_merged_topic)" />
    </node>
  </group>

</launch>
